### ISSUE TITLE
How to enable concurrent requests per replica?

### ISSUE THREAD

**Issue by robert4os** [Azure/MachineLearningNotebooks#910]
I have tried replica_max_concurrent_requests=2 which seems to enable queuing for the service, but no real concurrency.

A web service on a AKS cluster should be able to handle concurrent requests without a problem. 

Why is this limited?
Why should one contact MSFT technical support regarding this matter?

Thank you and best regards,
Robert

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a2c5068c-bf75-51ea-a8b3-a3356ba81b5c
* Version Independent ID: 9036f1f0-f669-bb28-74e7-3ea8dae6d7e9
* Content: [azureml.core.webservice.AksWebservice class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.webservice.akswebservice?view=azure-ml-py#deploy-configuration-autoscale-enabled-none--autoscale-min-replicas-none--autoscale-max-replicas-none--autoscale-refresh-seconds-none--autoscale-target-utilization-none--collect-model-data-none--auth-enabled-none--cpu-cores-none--memory-gb-none--enable-app-insights-none--scoring-timeout-ms-none--replica-max-concurrent-requests-none--max-request-wait-time-none--num-replicas-none--primary-key-none--secondary-key-none--tags-none--properties-none--description-none--gpu-cores-none--period-seconds-none--initial-delay-seconds-none--timeout-seconds-none--success-threshold-none--failure-threshold-none--namespace-none--token-auth-enabled-none--compute-target-name-none-)
* Content Source: [AzureML-Docset/stable/docs-ref-autogen/azureml-core/azureml.core.webservice.AksWebservice.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML-Docset/stable/docs-ref-autogen/azureml-core/azureml.core.webservice.AksWebservice.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @j-martens
* Microsoft Alias: **jmartens**

**Comment by GiftA-MSFT** [comment]
@robert4os we will review your feedback and get back to you shortly. Thanks.

**Comment by GiftA-MSFT** [comment]
@j-martens please can you help clarify why this setting shouldn't be changed unless instructed?

**Comment by j-martens** [comment]
@GiftA-MSFT - please assign to the product team. thank you! Please let us know what we can do in documentation once the answer is uncovered. thank you!

**Comment by GiftA-MSFT** [comment]
@robert4os here's the feedback I received from the product team:

"The services that utilize the AzureML base images do not support parallel request processing. This parameter is only for FPGA-enabled services or customers using their own inferencing stack. Also, for no code deployment (tensorflow, onnx, sklearn), python's global interpreter lock does not allow concurrent processing. That label controls the number of outstanding requests that the Azure ML Front End will route to a single container, but doesn't change anything about the underlying container "

Hope this helps. Thanks.
